en/developer/reference/articles/02-workflow/05-notification-templates.markdown

```diff
@@ -0,0 +1,287 @@
+| ${portalPermission}          |          | &#10004;  | [PortalPermission](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/service/permission/PortalPermission.html) |
+| ${entryClassPK}              | &#10004; |           | The primary key of the Entry Class           |
+| ${layoutPermission}          |          | &#10004;  | [LayoutPermission](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/service/permission/LayoutPermission.html) |
+| ${expandoTableLocalService}  |          | &#10004;  | [ExpandoTableLocalService](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/expando/kernel/service/ExpandoTableLocalService.html) |
+| ${localeUtil}                |          | &#10004;  | [LocaleUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/LocaleUtil.html) |
+| ${groupId}                   | &#10004; |           | The ID of the group                          |
+| ${portalUtil}                |          | &#10004;  | [PortalUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/PortalUtil.html) |
+| ${validator}                 |          | &#10004;  | [Validator](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/Validator.html) |
+| ${dateUtil}                  |          | &#10004;  | [DateUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/DateUtil.html) |
+| ${serviceLocator}            |          | &#10004;  | [ServiceLocator](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-impl/com/liferay/portal/template/ServiceLocator.html) |
+| ${serviceContext}            |          | &#10004;  | [ServiceContext](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/service/ServiceContext.html) |
+| ${jsonFactoryUtil}           |          | &#10004;  | [JSONFactoryUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/json/JSONFactoryUtil.html) |
+| ${stringUtil}                |          | &#10004;  | [StringUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/StringUtil.html) |
+| ${freeMarkerPortletPreferences} |       | &#10004;  | [TemplatePortletPreferences](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-impl/com/liferay/portal/template/TemplatePortletPreferences.html) |
+| ${dateFormats}              | &#10004;  |           | DESCRIBE |
+| ${dateFormatFactory}         |          | &#10004;  | [DateFormatFactory](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/DateFormatFactory.html) |
+| ${userPortraitURL}           | &#10004; |           | DESCRIBE |
```

Comment:
@rafaprax another one i could use help with :point_up_2:

Reply:
@rbohl This one is specific for BlogsEntries.  Can we remove this? What do you think?

Reply:
I'd instead say this: 

> Applies only to Blogs Entries. Returns a URL to the current User's avatar.

Of course I don't know if that's accurate.

Reply:
@rbohl,

>Applies only to Blogs Entries. Returns a URL to the current User's avatar.

It's ok to me.